### ISSUE TITLE
[naming-convention] camelCase should allow leading capital letters

### ISSUE THREAD

**Issue by snitin315** [typescript-eslint/typescript-eslint#4587]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

## Repro

[Playground](https://typescript-eslint.io/play#ts=4.5.2&sourceType=module&code=MYewdgzgLgBAggBwQOQIYFsCmMC8MBEUm0+A3EA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AQIYC2hA5lkgPYEBucRajIAXCHCivShAAaEEkpxcYhHC48+AkAF8gA&tsConfig=N4XyA)

### Code

```ts
const AppName = "test";
```

### Eslint config

```json
{
  "rules": {
    "@typescript-eslint/naming-convention": "error",
    "camelcase": "error"
  }
}
```

## Expected Result

No errors.

## Actual Result

<img width="827" alt="Screenshot 2022-02-22 at 3 57 06 PM" src="https://user-images.githubusercontent.com/46647141/155113564-71a9107c-22c3-4993-a640-50a10cfa19a9.png">

## Additional Info

Since `camelCase` rule is deprecated, I want to upgrade to `naming-convention` rule. It is very common for codebase to have both styles (`camelCase` and `CamelCase`), specially for react apps where component name starts with capital alphabet.

## Versions

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.12.1` |
| `@typescript-eslint/parser`        | `5.12.1` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.1.0` |
| `Env`                              | `web` |

**Comment by snitin315** [comment]
@Josh-Cena  Thank you, I will try `PascalCase` and close the ticket if all works fine.
